include/deal.II/dofs/dof_accessor.h

```diff
@@ -1970,82 +1955,110 @@ class DoFCellAccessor : public DoFAccessor<DoFHandlerType::dimension,
    */
 
   /**
-   * Return whether the p-refinement flag is set or not.
+   * Return the fe_index of the future finite element that this object
+   * will be assigned to after the triangulation changed (due to refinement
+   * and/or coarsening, or repartitioning). If no future finite element has
+   * been specified, the active one will be preserved and thus the fe_index
+   * of the active finite element will be returned.
    *
-   * @note The DoFHandler in use has to be of type hp::DoFHandler.
-   * An exception is thrown otherwise.
-   */
-  bool
-  p_refine_flag_set() const;
-
-  /**
-   * Flag the cell pointed to for p-refinement. This function is only allowed
-   * for active cells.
+   * @note Since degrees of freedom only exist on active cells for
+   * hp::DoFHandler (i.e., there is currently no implementation of multilevel
+   * hp::DoFHandler objects), it does not make sense to query future FE
+   * indices on non-active cells since they do not have finite element spaces
+   * associated with them without having any degrees of freedom. Consequently,
+   * this function will produce an exception when called on non-active cells.
    *
-   * @note The DoFHandler in use has to be of type hp::DoFHandler.
-   * An exception is thrown otherwise.
+   * @note When using parallel meshes, either through the
+   * parallel::shared::Triangulation or parallel::distributed::Triangulation
+   * classes, it is only allowed to call this function on locally
+   * owned or ghost cells. No information is available on artificial cells.
+   * Furthermore, @p future_fe_index information is exchanged from locally
+   * owned cells on one processor to other processors where they may be
+   * ghost cells, during the call to hp::DoFHandler::set_fe() and
+   * hp::DoFHandler::distribute_dofs(). See the documentation of hp::DoFHandler
+   * for more information.
    */
-  void
-  set_p_refine_flag() const;
+  unsigned int
+  future_fe_index() const;
 
   /**
-   * Clear the p-refinement flag.
+   * Set the fe_index of the future finite element that this object
+   * will be assigned to after the triangulation changed (due to refinement
+   * and/or coarsening, or repartitioning). A previously assigned future finite
+   * element will be overwritten.
```

Comment:
I would remove the same docu and instead write

> See future_fe_index() for more information.

Reply:
I agree.